modules/javafx.graphics/src/main/java/com/sun/prism/d3d/D3DTextureResource.java

```diff
@@ -38,6 +38,7 @@ public D3DTextureResource(D3DTextureData resource) {
     public void free() {
         if (resource != null) {
             resource.dispose();
+            resource = null;
```

Comment:
We do not set the resource to `null` for any classes like `D3DTextureResource`, `MTLTextureResource` or `ES2TextureResource`. But the actual reference to native resource is set to **0** in the `resource.dispose();` call ( i.e. in classes like, `D3DTextureData`, `MTLTextureData`, `ES2TextureData` ). So the actual dispose of resource is already guarded.

Setting null only for D3D may cause confusion, as to why is it not done for other pipelines.
So, I would recommend to remove, as current code is safe. Or
add the same for other pipelines as well.

Reply:
That might be a good idea (for all pipelines).

Reply:
It does make more sense, I noticed that we also often check `resource.isValid()` but we don't precede it with `resource == null` which would cause another potential NPE.

I'll revert this and add a comment in all backend-specific `*TextureResource` classes